.github/workflows/pr-check.yml

```diff
@@ -13,27 +13,100 @@ env:
     PNAME: loadgen
 
 jobs:
+    format_check:
+        runs-on: ubuntu-latest
+        steps:
+            - name: Checkout current repository
+              uses: actions/checkout@v4
+              with:
+                path: ${{ env.PNAME }}
+
+            - name: Checkout framework repository
+              uses: actions/checkout@v4
+              with:
+                repository: infinilabs/framework
+                path: framework
+
+            - name: Checkout framework-vendor
+              uses: actions/checkout@v4
+              with:
+                ref: main
+                repository: infinilabs/framework-vendor
+                path: vendor
+
+            - name: Set up go toolchain
+              uses: actions/setup-go@v5
+              with:
+                go-version: ${{ env.GO_VERSION }}
+                check-latest: false
+                cache: true
+
+            - name: Check go toolchain
+              run: go version
+              
+            - name: Run make format
+              shell: bash
+              run: |
+                echo Home path is $HOME
+                export WORKBASE=$HOME/go/src/infini.sh
+                export WORK=$WORKBASE/$PNAME
+
+                # for test workspace
+                mkdir -p $HOME/go/src/
+                ln -s $GITHUB_WORKSPACE $WORKBASE
+                
+                # check work folder
+                ls -lrt $WORKBASE/
+                ls -alrt $WORK
+
+                # for unit test
+                cd $WORK
+                echo Formating code at $PWD ...
+                make format
+                if [ $? -ne 0 ]; then
+                  echo "make format failed, please check make output"
+                  exit 1
+                fi
```

Comment:
I am not quite sure about this, but looks like if `make format` exits with a non-zero code, then the whole CI pipeline would fail, and this `if` statement won't be executed